module-api/src/main/java/com/pepponechoi/cinema/reservation/service/ReservationServiceImpl.java

```diff
@@ -1,8 +1,170 @@
+        }
+
+        // Seat의 크기가 2이상이면 Valid
+        if (seats.size() >= 2) {
+            Map<Character, Set<Integer>> setMap = new HashMap<>();
+
+            seats.forEach(seat -> {
+                setMap.computeIfAbsent(seat.getRowNo(), k -> new HashSet<>());
+                if (setMap.get(seat.getRowNo()).contains(seat.getColumnNo())) {
+                    ConflictException exception = new ConflictException();
+                    exception.setErrorCode(ConfliectErrorCode.CONFLICT);
+                    exception.setDetail("좌석은 중복될 수 없습니다.");
+                    throw exception;
+                }
+                setMap.get(seat.getRowNo()).add(seat.getColumnNo());
+            });
+
+            if (setMap.size() >= 2) {
+                BadRequestException exception = new BadRequestException();
+                exception.setErrorCode(BadRequestErrorCode.VALIDATION_FAILED);
+                exception.setDetail("2개 이상의 좌석은 같은 행에 있어야 합니다.");
+                throw exception;
+            }
+
+            setMap.forEach((rowNo, columnNoSet) -> {
+                List<Integer> seatRowNos = columnNoSet.stream().sorted().toList();
+
+                if (!IntStream.range(1, seatRowNos.size())
+                    .allMatch(index -> seatRowNos.get(index) - seatRowNos.get(index - 1) == 1)
+                ) {
+                    BadRequestException exception = new BadRequestException();
+                    exception.setErrorCode(BadRequestErrorCode.VALIDATION_FAILED);
+                    exception.setDetail("좌석은 이어져 있어야 합니다.");
+                    throw exception;
+                }
+            });
+        }
+
+        Reservation reservation = Reservation.of(user, seats, schedule, String.valueOf(user.getId()));
+
+        List<RLock> locks = new ArrayList<>();
+
+        try {
+            for (var seat : request.seats()) {
+                String seatLockKey =
+                    "seatLock:" + request.scheduleId() + ":" + seat.rowNo() + ":" + seat.columnNo();
+                RLock seatLock = redisson.getLock(seatLockKey);
+
+                if (!seatLock.tryLock(2, 1, TimeUnit.SECONDS)) {
+                    ConflictException exception = new ConflictException();
+                    exception.setDetail("좌석 예약 중 충돌이 발생했습니다. 다시 시도해 주세요.");
+                    throw exception;
+                }
+                locks.add(seatLock); // Save acquired locks
+            }
+
+            reservationRepository.save(reservation);
+
+            seats.forEach(seat -> {
+                seat.setReservation(reservation);
+            });
+
+            eventPublisher.publish(new ReservationCompletedEvent(
+                reservation.getId(),
+                reservation.getUser().getId()
+            ));
+        } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
+            ConflictException exception = new ConflictException();
+            exception.setDetail("록 획득 중 인터럽트 되었습니다.");
+        }
```

Comment:
좌석별 락을 모두 획득한 후에는 finally 블록을 통해 반드시 해제해주는 게 좋을 것 같습니다.
그렇지 않으면 일정 시간 동안 락이 유지되어 다른 사용자의 예약 요청이 차단될 수 있어요.

```suggestion
     
        } catch (InterruptedException e) {
            Thread.currentThread().interrupt();
            ConflictException exception = new ConflictException();
            exception.setDetail("록 획득 중 인터럽트 되었습니다.");
        }
        finally {
    for (RLock lock : locks) {
        if (lock.isHeldByCurrentThread()) {
            lock.unlock();
        }
    }
}
```

Reply:
해당 코드 반영하였습니다!